ygen/gogen.go

```diff
@@ -2019,85 +1981,94 @@ func generateGetListKey(buf *bytes.Buffer, s *Directory, nameMap map[string]*yan
 //	  type.
 // In the case that the list has multiple keys, the type generated as the key of the list is returned.
 // If errors are encountered during the type generation for the list, the error is returned.
-func yangListFieldToGoType(listField *yang.Entry, listFieldName string, parent *Directory, goStructElements map[string]*Directory, gogen *GoLangMapper) (string, *generatedGoMultiKeyListStruct, *generatedGoListMethod, error) {
+func yangListFieldToGoType(listField *NodeDetails, listFieldName string, parent *ParsedDirectory, goStructElements map[string]*ParsedDirectory) (string, *generatedGoMultiKeyListStruct, *generatedGoListMethod, error) {
 	// The list itself, since it is a container, has a struct associated with it. Retrieve
 	// this from the set of Directory structs for which code (a Go struct) will be
 	//  generated such that additional details can be used in the code generation.
-	listElem, ok := goStructElements[listField.Path()]
+	listElem, ok := goStructElements[listField.YANGDetails.Path]
 	if !ok {
-		return "", nil, nil, fmt.Errorf("struct for %s did not exist", listField.Path())
+		return "", nil, nil, fmt.Errorf("struct for %s did not exist", listField.YANGDetails.Path)
 	}
 
-	// Find the name of the struct that refers to the list provided as input. The YANGCodeGenerator
-	// instance stores this information. It is populated by structName which always runs prior to
-	// this function being called (as all mappable entities, which includes lists, have been found.
-	// Thus, in the case that this struct does not have a known name, then code cannot be generated
-	// for it, and hence we skip the element.
-	listName, ok := gogen.uniqueDirectoryNames[listField.Path()]
-	if !ok {
-		return "", nil, nil, fmt.Errorf("list element %s did not have a resolved name", listField.Path())
-	}
-
-	if listElem.ListAttr == nil || len(listElem.ListAttr.Keys) == 0 {
+	if len(listElem.ListKeys) == 0 {
 		// Keyless list therefore represent this as a slice of pointers to
 		// the struct that represents the list element itself.
-		return fmt.Sprintf("[]*%s", listName), nil, nil, nil
+		return fmt.Sprintf("[]*%s", listElem.Name), nil, nil, nil
 	}
 
 	var listType string
 	var multiListKey *generatedGoMultiKeyListStruct
 	var listKeys []goStructField
 	var listKeyStructName string
 
-	// Key name elements are ordered per Section 7.8.2 of RFC6020. Rely on this
-	// fact for determisitic ordering in output code and rendering.
-	keyElemNames := strings.Fields(listField.Key)
+	shortestPath := func(ss [][]string) [][]string {
```

Comment:
Could we add a comment here (or at the call site) since I think it took me a second to realise that the shortest path must be the key's path because the key cannot be within a descendent container.

Reply:
Done.